google/resource_container_node_pool.go

```diff
@@ -85,6 +87,34 @@ func resourceContainerNodePoolCreate(d *schema.ResourceData, meta interface{}) e
 		InitialNodeCount: int64(nodeCount),
 	}
 
+	if v, ok := d.GetOk("node_config"); ok {
+		nodeConfigs := v.([]interface{})
+		if len(nodeConfigs) > 1 {
+			return fmt.Errorf("Cannot specify more than one node_config.")
+		}
+		nodeConfig := nodeConfigs[0].(map[string]interface{})
+
+		nodePool.Config = &container.NodeConfig{}
+
+		if v, ok = nodeConfig["machine_type"]; ok {
+			nodePool.Config.MachineType = v.(string)
+		}
+
+		if v, ok = nodeConfig["disk_size_gb"]; ok {
+			nodePool.Config.DiskSizeGb = int64(v.(int))
+		}
+
+		if v, ok := nodeConfig["oauth_scopes"]; ok {
+			scopesList := v.([]interface{})
+			scopes := []string{}
+			for _, v := range scopesList {
+				scopes = append(scopes, canonicalizeServiceScope(v.(string)))
+			}
+
+			nodePool.Config.OauthScopes = scopes
+		}
+	}
```

Comment:
What about the other fields? (local_ssd_count, service_account, metadata, image_type, labels, tags)